jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/LogStreamFilterActorTest.kt

```diff
@@ -0,0 +1,184 @@
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs
+
+import com.intellij.testFramework.ProjectRule
+import com.intellij.ui.table.TableView
+import com.intellij.util.ui.ListTableModel
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.ClosedSendChannelException
+import kotlinx.coroutines.debug.junit4.CoroutinesTimeout
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineScope
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.After
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsAsyncClient
+import software.amazon.awssdk.services.cloudwatchlogs.model.FilterLogEventsRequest
+import software.amazon.awssdk.services.cloudwatchlogs.model.FilterLogEventsResponse
+import software.amazon.awssdk.services.cloudwatchlogs.model.FilteredLogEvent
+import software.aws.toolkits.jetbrains.core.MockClientManagerRule
+import software.aws.toolkits.jetbrains.utils.waitForFalse
+import software.aws.toolkits.jetbrains.utils.waitForModelToBeAtLeast
+import software.aws.toolkits.resources.message
+import java.time.Duration
+import java.util.concurrent.CompletableFuture
+
+// ExperimentalCoroutinesApi is needed for TestCoroutineScope
+@ExperimentalCoroutinesApi
+class LogStreamFilterActorTest {
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    @JvmField
+    @Rule
+    val mockClientManagerRule = MockClientManagerRule(projectRule)
+
+    @JvmField
+    @Rule
+    val timeout = CoroutinesTimeout.seconds(15)
+
+    private val testCoroutineScope: TestCoroutineScope = TestCoroutineScope()
+
+    @After
+    fun after() {
+        testCoroutineScope.cleanupTestCoroutines()
+    }
+
+    @Test
+    fun modelIsPopulated() {
+        val mockClient = mockClientManagerRule.create<CloudWatchLogsAsyncClient>()
+        whenever(mockClient.filterLogEvents(Mockito.any<FilterLogEventsRequest>()))
+            .thenReturn(
+                CompletableFuture.completedFuture(
+                    FilterLogEventsResponse
+                        .builder()
+                        .events(FilteredLogEvent.builder().message("message").build())
+                        .build()
+                )
+            )
+        val tableModel = ListTableModel<LogStreamEntry>()
+        val table = TableView<LogStreamEntry>(tableModel)
+        val actor = LogStreamFilterActor(projectRule.project, table, "abc", "def")
+        runBlocking {
+            actor.channel.send(LogStreamActor.Message.LOAD_INITIAL_FILTER("filter query"))
+            tableModel.waitForModelToBeAtLeast(1)
+        }
+        assertThat(tableModel.items.size).isOne()
+        assertThat(tableModel.items.first().message).isEqualTo("message")
```

Comment:
should we also test that timestamp is populated?

Reply:
That would make sense, added to the tests